pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go

```diff
@@ -121,17 +122,44 @@ func (c *ClusterRoleAggregationController) syncClusterRole(key string) error {
 		return nil
 	}
 
-	// we need to update
-	clusterRole := sharedClusterRole.DeepCopy()
-	clusterRole.Rules = nil
-	for _, rule := range newPolicyRules {
-		clusterRole.Rules = append(clusterRole.Rules, *rule.DeepCopy())
+	clusterRoleApply := rbacv1ac.ClusterRole(sharedClusterRole.Name).
+		WithRules(toApplyPolicyRules(newPolicyRules)...)
+
+	opts := metav1.ApplyOptions{FieldManager: "clusterrole-aggregation-controller", Force: true}
+	_, err = c.clusterRoleClient.ClusterRoles().Apply(context.TODO(), clusterRoleApply, opts)
+	if errors.IsUnsupportedMediaType(err) { // TODO: Remove this fallback in 1.22 or later.
```

Comment:
also skip if the feature is explicitly disabled, right?

```suggestion
	opts := metav1.ApplyOptions{FieldManager: "clusterrole-aggregation-controller", Force: true}
	_, err = c.clusterRoleClient.ClusterRoles().Apply(context.TODO(), clusterRoleApply, opts)
	if errors.IsUnsupportedMediaType(err) { // TODO: Remove this fallback at least one release after ServerSideApply GA
```